core/store/hashtree/binprefix/mod.go

```diff
@@ -60,9 +64,35 @@ func NewMerkleTree(db kv.DB, nonce Nonce) *MerkleTree {
 	}
 }
 
+// Load tries to read the bucket and scan it for existing leafs and populate the
+// tree with them.
+func (t *MerkleTree) Load() error {
+	t.Lock()
+	defer t.Unlock()
+
+	return t.doUpdate(func(tx kv.WritableTx) error {
+		bucket := tx.GetBucket(t.bucket)
+
+		err := t.tree.Load(bucket)
+		if err != nil {
+			return xerrors.Errorf("failed to load: %v", err)
+		}
+
+		err = t.tree.Update(t.hashFactory, bucket)
```

Comment:
I find  `t.tree.Load(bucket)` and `t.tree.Update(...)` to be very  meaningless. But I guess it's hard to make that code more explicit at this stage.

Reply:
`Update` => `CalculateRoot`
`Load` => `LaodFromDisk`

That might help a bit.